File: src/lib/software/bitwarden.py
from src.lib.software import BaseSoftware

class Bitwarden(BaseSoftware):
    def __init__(self):
        super().__init__()

        self.download_name = 'Bitwarden-Installer.exe'
        self.silent_install_args = ['/S']

    def resolve_download_url(self):
        self.url_resolved.emit('https://vault.bitwarden.com/download/?app=desktop&platform=windows&variant=exe')

File: packages/app/src/services/software/definitions/bitwarden.ts
import { ok } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'bitwarden-desktop';
	public name = 'Bitwarden';
	public category = [SoftwareCategory.Security];
	public downloadName = 'Bitwarden-Installer.exe';
	public icon = 'bitwarden.png';
	public homepage = 'https://bitwarden.com';

	public async resolveDownloadUrl() {
		return ok('https://vault.bitwarden.com/download/?app=desktop&platform=windows&variant=exe');
	}
}
